app/src/main/java/com/omkarmoghe/pokemap/helpers/RemoteImageLoader.java

```diff
@@ -2,46 +2,67 @@
 
 import android.content.Context;
 import android.graphics.Bitmap;
+import android.graphics.Matrix;
+import android.graphics.RectF;
 import android.graphics.drawable.Drawable;
+import android.util.LruCache;
 
 import com.bumptech.glide.Glide;
 import com.bumptech.glide.load.engine.DiskCacheStrategy;
 import com.bumptech.glide.request.animation.GlideAnimation;
 import com.bumptech.glide.request.target.SimpleTarget;
 
-import java.util.HashMap;
-import java.util.Map;
-
 /**
  * Cache wrapper around the Glide image loader.
  * <p>
  * Created by aronhomberg on 26.07.16.
  */
 public class RemoteImageLoader {
+    private static final String TAG = "RemoteImageLoader";
+    private static final int MAX_SIZE = (int) (Runtime.getRuntime().maxMemory() / 1024);
+    private static final LruCache<String, Bitmap> BITMAP_CACHE = new LruCache<String, Bitmap>((int) (MAX_SIZE /8)){
+        @Override
+        protected int sizeOf(String key, Bitmap value) {
+            return value.getByteCount() / 1024;
+        }
+    };
 
-    private static Map<String, Bitmap> bitmapCache = new HashMap<>();
-
-    public static void load(final String url, int pxWidth, int pxHeight,
+    public static void load(final String url, final int pxWidth, final int pxHeight,
                             Drawable placeholderDrawable, Context context, final Callback onFetch) {
-        // ultra-fast cache reply
-        if (bitmapCache.containsKey(url)) {
-            onFetch.onFetch(bitmapCache.get(url));
+        Bitmap bitmap = BITMAP_CACHE.get(url);
+        if(bitmap != null){
+            bitmap = scaleBitmap(bitmap, pxWidth, pxHeight);
+            onFetch.onFetch(bitmap);
         } else {
             Glide.with(context).load(url)
                     .asBitmap()
                     .skipMemoryCache(false)
                     .placeholder(placeholderDrawable)
-                    .diskCacheStrategy(DiskCacheStrategy.ALL)
-                    .into(new SimpleTarget<Bitmap>(pxWidth, pxHeight) {
+                    .diskCacheStrategy(DiskCacheStrategy.SOURCE)
+                    .into(new SimpleTarget<Bitmap>() {
                         @Override
                         public void onResourceReady(Bitmap bitmap, GlideAnimation anim) {
-                            bitmapCache.put(url, bitmap);
+                            addBitmapToMemoryCache(url, bitmap);
+                            bitmap = scaleBitmap(bitmap, pxWidth, pxHeight);
```

Comment:
Let's just store the scaled bitmap in the cache instead of doing the scaling each time